pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1293,23 +1293,19 @@ self: super: {
   # Fails with "supports custom headers"
   Spock-core = dontCheck super.Spock-core;
 
-  # Needed by Hasura  1.3.1
-  dependent-map_0_2_4_0 = super.dependent-map_0_2_4_0.override {
-    dependent-sum = self.dependent-sum_0_4;
-  };
-
-  # Hasura 1.3.1
-  # Because of ghc-heap-view, profiling needs to be disabled.
-  graphql-engine = disableLibraryProfiling( overrideCabal (super.graphql-engine.override {
-     immortal = self.immortal_0_2_2_1;
-     dependent-map = self.dependent-map_0_2_4_0;
-     dependent-sum = self.dependent-sum_0_4;
+  # hasura packages need some extra care
+  graphql-engine = overrideCabal (super.graphql-engine.override {
+    immortal = self.immortal_0_2_2_1;
   }) (drv: {
-     # version in cabal file is invalid
-     version = "1.3.1-beta1";
-     # hasura needs VERSION env exported during build
-     preBuild = "export VERSION=1.3.1-beta1";
-  }));
+    patches = [ ./patches/graphql-engine-mapkeys.patch ];
+    isLibrary = false;
```

Comment:
Maybe add a note here why this is needed? I was a bit confused looking at the code, but given that graphlql-engine is a server this shouldn't matter much.

Reply:
why the `isLibrary = false` ? actually I don't know, it just fails if I have it enabled

Reply:
This should be fixed by deleting all the [comments above this line](https://github.com/hasura/graphql-engine/blob/5990ca403bf6e7dd66081720c329e513f0624fb6/server/src-lib/Hasura/Backends/BigQuery/Types.hs#L385-L400) via a patch or similar, I think at least.

Reply:
ah, nice, while I have you here. Maybe you know whats wrong with this error: my patch fixes it, but I'm not sure it does make any sense (since the local defined mapKeys is now unused):
```
src-lib/Data/HashMap/Strict/Extended.hs:10:5: error:
    Ambiguous occurrence ‘mapKeys’
    It could refer to
       either ‘M.mapKeys’,
              imported from ‘Data.HashMap.Strict’ at src-lib/Data/HashMap/Strict/Extended.hs:20:1-41
              (and originally defined in ‘Data.HashMap.Internal’)
           or ‘Data.HashMap.Strict.Extended.mapKeys’,
              defined at src-lib/Data/HashMap/Strict/Extended.hs:69:1
   |
10 |   , mapKeys
   |     ^^^^^^^
   ```

Reply:
```diff
- import           Data.HashMap.Strict as M
+ import qualified Data.HashMap.Strict as M
```

in that file should fix that without deleting the export.

Reply:
sadly it doesn't seem to be that easy:
```
src-lib/Data/HashMap/Strict/Extended.hs:25:14: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
25 | catMaybes :: HashMap k (Maybe v) -> HashMap k v
   |              ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:25:37: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
25 | catMaybes :: HashMap k (Maybe v) -> HashMap k v
   |                                     ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:28:56: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
28 | fromListOn :: (Eq k, Hashable k) => (v -> k) -> [v] -> HashMap k v
   |                                                        ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:32:51: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
32 | unionsAll :: (Eq k, Hashable k, Foldable t) => t (HashMap k v) -> HashMap k (NonEmpty v)
   |                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:32:67: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
32 | unionsAll :: (Eq k, Hashable k, Foldable t) => t (HashMap k v) -> HashMap k (NonEmpty v)
   |                                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:41:65: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
41 | groupOn :: (Eq k, Hashable k, Foldable t) => (v -> k) -> t v -> HashMap k [v]
   |                                                                 ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:44:67: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
44 | groupOnNE :: (Eq k, Hashable k, Foldable t) => (v -> k) -> t v -> HashMap k (NonEmpty v)
   |                                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:47:77: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
47 | differenceOn :: (Eq k, Hashable k, Foldable t) => (v -> k) -> t v -> t v -> HashMap k v
   |                                                                             ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:51:34: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
51 | lpadZip :: (Eq k, Hashable k) => HashMap k a -> HashMap k b -> HashMap k (Maybe a, b)
   |                                  ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:51:49: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
51 | lpadZip :: (Eq k, Hashable k) => HashMap k a -> HashMap k b -> HashMap k (Maybe a, b)
   |                                                 ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:51:64: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
51 | lpadZip :: (Eq k, Hashable k) => HashMap k a -> HashMap k b -> HashMap k (Maybe a, b)
   |                                                                ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:68:51: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
68 | mapKeys :: (Ord k2, Hashable k2) => (k1 -> k2) -> HashMap k1 a -> HashMap k2 a
   |                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:68:67: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
68 | mapKeys :: (Ord k2, Hashable k2) => (k1 -> k2) -> HashMap k1 a -> HashMap k2 a
   |                                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:78:70: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
78 | unionsWith :: (Foldable f, Hashable k, Ord k) => (a -> a -> a) -> f (HashMap k a) -> HashMap k a
   |                                                                      ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:78:86: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
78 | unionsWith :: (Foldable f, Hashable k, Ord k) => (a -> a -> a) -> f (HashMap k a) -> HashMap k a
   |                                                                                      ^^^^^^^
   ```

Reply:
Then let's make it

```diff
- import           Data.HashMap.Strict as M
+ import qualified Data.HashMap.Strict as M
+ import           Data.HashMap.Strict (HashMap ())
```

Reply:
already had a similiar idea:
```
rc-lib/Data/HashMap/Strict/Extended.hs:30:16: error:
    • Variable not in scope: fromList :: [(k, v)] -> HashMap k v
    • Perhaps you meant one of these:
        ‘M.fromList’ (imported from Data.HashMap.Strict),
        ‘fromListOn’ (line 30)
      Perhaps you want to add ‘fromList’ to one of these import lists:
        ‘Data.List.NonEmpty’ (src-lib/Data/HashMap/Strict/Extended.hs:23:1-52)
        ‘Data.HashMap.Strict’ (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46)
   |
30 | fromListOn f = fromList . Prelude.map (\v -> (f v, v))
   |                ^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:34:31: error:
    • Variable not in scope:
        unionWith
          :: (a1 -> a1 -> a1)
             -> HashMap k (NonEmpty v)
             -> HashMap k (NonEmpty v)
             -> HashMap k (NonEmpty v)
    • Perhaps you meant one of these:
        ‘M.unionWith’ (imported from Data.HashMap.Strict),
        ‘unionsWith’ (line 80)
      Perhaps you want to add ‘unionWith’ to the import list
      in the import of ‘Data.HashMap.Strict’
      (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46).
   |
34 | unionsAll = F.foldl' (\a b -> unionWith (<>) a (fmap (:|[]) b)) M.empty
   |                               ^^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:70:13: error:
    • Variable not in scope: fromList :: b1 -> HashMap k2 a
    • Perhaps you meant one of these:
        ‘M.fromList’ (imported from Data.HashMap.Strict),
        ‘fromListOn’ (line 30)
      Perhaps you want to add ‘fromList’ to one of these import lists:
        ‘Data.List.NonEmpty’ (src-lib/Data/HashMap/Strict/Extended.hs:23:1-52)
        ‘Data.HashMap.Strict’ (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46)
   |
70 | mapKeys f = fromList . foldrWithKey (\k x xs -> (f k, x) : xs) []
   |             ^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:70:24: error:
    • Variable not in scope:
        foldrWithKey
          :: (k1 -> b0 -> [(k2, b0)] -> [(k2, b0)])
             -> [a0] -> HashMap k1 a -> b1
    • Perhaps you meant one of these:
        ‘M.foldrWithKey’ (imported from Data.HashMap.Strict),
        ‘M.foldlWithKey’ (imported from Data.HashMap.Strict),
        ‘M.foldrWithKey'’ (imported from Data.HashMap.Strict)
      Perhaps you want to add ‘foldrWithKey’ to the import list
      in the import of ‘Data.HashMap.Strict’
      (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46).
   |
70 | mapKeys f = fromList . foldrWithKey (\k x xs -> (f k, x) : xs) []
   |                        ^^^^^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:81:15: error:
    • Variable not in scope:
        unionWith
          :: (a -> a -> a) -> HashMap k a -> HashMap k a -> HashMap k a
    • Perhaps you meant one of these:
        ‘M.unionWith’ (imported from Data.HashMap.Strict),
        ‘unionsWith’ (line 80)
      Perhaps you want to add ‘unionWith’ to the import list
      in the import of ‘Data.HashMap.Strict’
      (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46).
   |
81 |   = F.foldl' (unionWith f) empty ts
   |               ^^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:81:28: error:
    • Variable not in scope: empty :: HashMap k a
    • Perhaps you meant one of these:
        ‘M.empty’ (imported from Data.HashMap.Strict),
        ‘mempty’ (imported from Prelude)
      Perhaps you want to add ‘empty’ to the import list in the import of
      ‘Data.HashMap.Strict’
      (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46).
   |
81 |   = F.foldl' (unionWith f) empty ts
   |                            ^^^^^
   ```

Reply:
So I will probably just drop mapKeys again. since its copied in anyway

Reply:
Okay, this should work though:

```diff
- import           Data.HashMap.Strict as M
+ import           Data.HashMap.Strict as M hiding (mapKeys)
```

Reply:
ok, that worked, now there is another weird error:
```
src-lib/Hasura/Backends/MySQL/Meta.hs:198:22: error:
    parse error on input ‘-- ^ primay key’
    |
198 |         "PRI" -> PRI -- ^ primay key
    |                      ^^^^^^^^^^^^^^^
```

why do all this errors appear? does upstream build with a different haskell version?

Reply:
seems to be a haddock error, I will disable haddock for now

Reply:
Sounds sensible, no clue what upstream is doing…

Reply:
seems like upstream catched up with haddock :D
https://github.com/hasura/graphql-engine/commit/c1bdc993346c3690ed6fe708cf76959e508b24c6